File: Game Commons/src/de/gymdon/inf1315/game/Building.java
package de.gymdon.inf1315.game;

public abstract class Building extends GameObject {
    boolean[] options = new boolean[] { false, false, false, false, false, true, false };
    public int income;
    
    public abstract void occupy(Player p);
    
    @Override
    public boolean[] clicked(int phase) {
	if (phase != 0) {
	    options[5] = false;
	} else {
	    options[5] = true;
	}

	return options;
    }
    
    public int getIncome()
    {
	return income;
    }
}

File: Game Commons/src/de/gymdon/inf1315/game/Building.kt
package de.gymdon.inf1315.game

abstract class Building : GameObject() {
    var income: Int = 0
    override var options = booleanArrayOf(false, false, false, false, false, true, false)

    abstract fun occupy(p: Player)

    override fun clicked(phase: Int): BooleanArray {
        options[5] = phase == 0

        return options
    }
}
